dev/release/verify-release-candidate.sh

```diff
@@ -411,7 +411,8 @@ if [ "$ARTIFACT" == "source" ]; then
   cd ${DIST_NAME}
 
   rm -r testing
-  git clone https://github.com/apache/arrow-testing.git testing
+  git clone https://github.com/apache/arrow-testing.git
+  export ARROW_TEST_DATA=$PWD/arrow-testing/data
```

Comment:
If we don't use `${DIST_NAME}/testing`, `rm -r testing` is needless. And we don't put arrow-testing under `${DIST_NAME}/`.

How about this?

```diff
diff --git a/dev/release/verify-release-candidate.sh b/dev/release/verify-release-candidate.sh
index c8c631c4..4283b655 100755
--- a/dev/release/verify-release-candidate.sh
+++ b/dev/release/verify-release-candidate.sh
@@ -403,6 +403,9 @@ if [ "$ARTIFACT" == "source" ]; then
   TEST_JAVA=$((${TEST_JAVA} + ${TEST_INTEGRATION}))
   TEST_JS=$((${TEST_JS} + ${TEST_INTEGRATION}))
 
+  git clone https://github.com/apache/arrow-testing.git
+  export PARQUET_TEST_DATA=$PWD/arrow-testing/data
+
   git clone https://github.com/apache/parquet-testing.git
   export PARQUET_TEST_DATA=$PWD/parquet-testing/data
 
@@ -410,9 +413,6 @@ if [ "$ARTIFACT" == "source" ]; then
   tar xvzf ${DIST_NAME}.tar.gz
   cd ${DIST_NAME}
 
-  rm -r testing
-  git clone https://github.com/apache/arrow-testing.git testing
-
   if [ ${TEST_JAVA} -gt 0 ]; then
     test_package_java
   fi
```